File: 0060-permutation-sequence/0060-permutation-sequence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
    let numbers = [];
    let factorial = 1;
    for (let i = 1; i <= n; i++) {
        numbers.push(i);
        factorial *= i;
    }
    
    k--; // Convert k to zero-based index
    let result = '';
    
    for (let i = 0; i < n; i++) {
        factorial = Math.floor(factorial / (n - i));
        let index = Math.floor(k / factorial);
        result += numbers[index];
        numbers.splice(index, 1);
        k %= factorial;
    }
    
    return result;
};

File: 0060-permutation-sequence/0060-permutation-sequence.rs
impl Solution {
    pub fn get_permutation(n: i32, k: i32) -> String {
        let n = n as usize;
        let mut numbers: Vec<i32> = (1..=n as i32).collect();
        let mut factorial = (1..=n).product::<usize>();
        let mut k = k as usize - 1; // Convert k to zero-based index
        let mut result = String::new();
        
        for i in 0..n {
            factorial /= n - i;
            let index = k / factorial;
            result.push_str(&numbers[index].to_string());
            numbers.remove(index);
            k %= factorial;
        }
        
        result
    }
}
